packages/mobile/src/screens/favorites-screen/TracksTab.tsx

```diff
@@ -90,27 +89,25 @@ export const TracksTab = () => {
 
   const isLoading = savedTracksStatus !== Status.SUCCESS
 
-  const debouncedFetchSavesOnline = useMemo(() => {
+  const debouncedFetchSaves = useMemo(() => {
     return debounce((filterVal) => {
-      dispatch(fetchSaves(filterVal, '', '', 0, FETCH_LIMIT))
+      dispatch(fetchSavesAction(filterVal, '', '', 0, FETCH_LIMIT))
     }, 500)
   }, [dispatch])
 
-  const fetchSavesOnline = useCallback(() => {
-    debouncedFetchSavesOnline(filterValue)
-  }, [debouncedFetchSavesOnline, filterValue])
-
-  useOfflineFavoritesLineup(fetchSavesOnline)
+  const fetchSaves = useCallback(() => {
+    debouncedFetchSaves(filterValue)
+  }, [debouncedFetchSaves, filterValue])
 
   useEffect(() => {
+    // Need to fetch saves when the filterValue (by way of fetchSaves) changes
     if (isReachable) {
-      fetchSavesOnline()
+      fetchSaves()
     }
-  }, [isReachable, fetchSavesOnline])
+  }, [isReachable, fetchSaves])
 
-  const savedTrackUids: string[] = useSelector((state) => {
-    return getSavedTracksLineup(state).entries.map(({ uid }) => uid)
-  }, isEqual)
+  const { entries } = useFavoritesLineup(fetchSaves)
```

Comment:
so if im understanding, the fetchSaves whether called in this method or above will ultimately update the entries we select and return in this hook? makes sense but yeah a bit annoying we have to do a `useEffect` here as well. so could we potentially just rerun the `fetchSaves` call in the `useFavoritesLineup` when fetchSaves itself is a new function? maybe that's even more complicated. tldr im fine with this, but just asking qs

Reply:
Yeah that's the path I was going down but it means pulling all the pagination and filtering into the hook as well. So then the hook would accept the filterValue, and return the lineup along with a fetchMore function

Reply:
Otherwise there would still be a fetchMoreSaves in the component, so the fetch logic would be split into two places

Reply:
Rerunning fetchSaves when it changes becomes complicated bc we usually only want to run it on reachability change. I think passing in filterValue and moving all the pagination and filtering into the hook would be cleaner. I'll think on it some more